crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -2317,7 +2330,7 @@ impl<'src> Parser<'src> {
         }
         helpers::set_expr_ctx(&mut target, ExprContext::Store);
 
-        let value = self.parse_conditional_expression_or_higher(AllowStarredExpression::No);
+        let value = self.parse_conditional_expression_or_higher(ExpressionContext::default());
```

Comment:
Should we inherit the context here to avoid `for a:= in b` to eat the `in`?

Reply:
I don't think that's required at this point because the target value of a `for` statement uses the `expressions` (`parse_expression_list`) grammar rule which will stop at `:=` token as it doesn't parse a named expression. Maybe something to lookout for when working on error recovery.